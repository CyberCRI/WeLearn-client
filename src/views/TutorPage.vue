<script setup lang="ts">
import { ref } from 'vue';
import FirstStep from '@/components/tutor/FirstStep.vue';
import SecondStep from '@/components/tutor/SecondStep.vue';
import ThirdStep from '@/components/tutor/ThirdStep.vue';
import StepsIndicator from '@/components/tutor/StepsIndicator.vue';

const step = ref(1);
const setStep = (val) => {
  step.value = val;
};

const handleSearch = () => {
  console.log('handle search');
  step.value = step.value + 1;
};

const handleCreateSyllabus = () => {
  console.log('create syllabus');
  step.value = step.value + 1;
};

const stepToAction = {
  1: handleSearch,
  2: handleCreateSyllabus
};
</script>
<template>
  <div class="content-centered-wrapper">
    <StepsIndicator :step="step" :setStep="setStep" stepsLength="3" />
    <div class="layout-flex">
      <div class="flex-wrap" :class="{ shrink: step === 3 }">
        <FirstStep :disabled="step > 1" v-if="step >= 1" />
        <SecondStep :disabled="step > 2" :visible="step >= 2" />
      </div>
      <ThirdStep :visible="step >= 3" />
    </div>
    <div class="actions">
      <button class="button" v-if="step > 1" @click="step = step - 1">previous</button>
      <button class="button" v-if="step < 3" @click="stepToAction[step]()">next</button>
    </div>
  </div>
</template>

<style scoped>
.content-centered-wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1rem 2rem;
  height: 100%;
  overflow: hidden;
}

.actions {
  display: flex;
  gap: 1rem;
}

.layout-flex {
  display: flex;
  gap: 2rem;
  max-height: 85%;
  padding: 1rem 0rem;
}

.flex-wrap {
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  min-width: 20%;
  flex-grow: 1;
  flex-shrink: 1;
  flex-basis: 100%;
  transition: flex-grow 0s;
  height: 100%;
}

.flex-wrap.shrink {
  flex-grow: 0.7;
  flex-basis: 10%;
  & > * {
    flex-basis: 100%;
    transition: flex-grow 0s;
  }
}
</style>
