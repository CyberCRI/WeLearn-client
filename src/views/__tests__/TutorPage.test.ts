import { describe, it, expect, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';
import TutorPage from '../TutorPage.vue';
import { createPinia, setActivePinia } from 'pinia';

describe('TutorPage', () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });
  it('renders properly', () => {
    const wrapper = mount(TutorPage);
    expect(wrapper.element).toMatchSnapshot();
  });
});
