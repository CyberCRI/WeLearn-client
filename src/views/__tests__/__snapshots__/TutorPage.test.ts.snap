// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TutorPage > renders properly 1`] = `
<div
  class="content-centered-wrapper"
  data-v-8a8621cb=""
>
  <div
    class="is-flex"
    data-v-050e406a=""
    data-v-8a8621cb=""
  >
    
    <p
      class="step-inidicator active"
      data-v-050e406a=""
    >
      1
    </p>
    <p
      class="step-inidicator disabled"
      data-v-050e406a=""
    >
      2
    </p>
    <p
      class="step-inidicator disabled"
      data-v-050e406a=""
    >
      3
    </p>
    
  </div>
  <!--v-if-->
  <div
    class="layout-flex"
    data-v-8a8621cb=""
  >
    <div
      class="flex-wrap"
      data-v-8a8621cb=""
    >
      <div
        class="wrapper"
        data-test="fist-step"
        data-v-8a8621cb=""
        data-v-e090ef8d=""
      >
        <h1
          class="title is-4"
          data-v-e090ef8d=""
        >
          tutor.firstStep.title
        </h1>
        <p
          class="subtitle is-6"
          data-v-e090ef8d=""
        >
          tutor.firstStep.description
        </p>
        <div
          class="is-flex is-flex-wrap-wrap mb-4 descriptions"
          data-v-e090ef8d=""
        >
          <div
            class="description"
            data-v-e090ef8d=""
          >
            <label
              data-v-e090ef8d=""
              for="cursus-title"
            >
              tutor.firstStep.cursusTitleLabel
            </label>
            <input
              class="input decription"
              data-v-e090ef8d=""
              id="cursus-title"
              placeholder="tutor.firstStep.cursusTitlePlaceholder"
              type="text"
            />
          </div>
          <div
            class="description"
            data-v-e090ef8d=""
          >
            <label
              data-v-e090ef8d=""
              for="cursus-level"
            >
              tutor.firstStep.cursusLevelLabel
            </label>
            <input
              class="input decription"
              data-v-e090ef8d=""
              id="cursus-level"
              placeholder="tutor.firstStep.cursusLevelPlaceholder"
              type="text"
            />
          </div>
          <div
            class="description"
            data-v-e090ef8d=""
          >
            <label
              data-v-e090ef8d=""
              for="cursus-duration"
            >
              tutor.firstStep.cursusDurationLabel
            </label>
            <input
              class="input decription"
              data-v-e090ef8d=""
              id="cursus-duration"
              placeholder="tutor.firstStep.cursusDurationPlaceholder"
              type="text"
            />
          </div>
        </div>
        <div
          data-v-e090ef8d=""
        >
          <div
            class="input-group is-flex is-flex-direction-column"
            data-v-e090ef8d=""
          >
            <div
              class="is-flex is-flex-direction-row"
              data-v-e090ef8d=""
              id="first"
            >
              <input
                accept="application/pdf, text/*"
                class="input"
                data-v-e090ef8d=""
                placeholder="Enter the name of the file"
                type="file"
              />
              <button
                class="button"
                data-v-e090ef8d=""
              >
                x
              </button>
            </div>
          </div>
          <button
            class="button"
            data-v-e090ef8d=""
          >
             + 
          </button>
          <!--v-if-->
        </div>
      </div>
      <div
        class="wrapper"
        data-test="second-step"
        data-v-8a8621cb=""
        data-v-ebcfbade=""
      >
        <h1
          class="title is-4"
          data-v-ebcfbade=""
        >
          tutor.secondStep.title
        </h1>
        <p
          class="subtitle is-6"
          data-v-ebcfbade=""
        >
          tutor.secondStep.description
        </p>
        <div
          class="sources"
          data-v-ebcfbade=""
        >
          
          
        </div>
      </div>
    </div>
    <div
      class="wrapper"
      data-test="third-step"
      data-v-7bf7679d=""
      data-v-8a8621cb=""
    >
      <h1
        class="title is-4"
        data-v-7bf7679d=""
      >
        tutor.thirdStep.title
      </h1>
      <p
        class="subtitle is-6"
        data-v-7bf7679d=""
      >
        tutor.thirdStep.description
      </p>
      <!-- syllabus -->
      
      
    </div>
  </div>
  <div
    class="actions"
    data-v-8a8621cb=""
  >
    <!--v-if-->
    <button
      class="button"
      data-v-8a8621cb=""
    >
      next
    </button>
  </div>
</div>
`;
